pkg/cmd/create/create.go

```diff
@@ -751,6 +783,28 @@ func (o *Options) watchNamespaceStart() error {
 	return nil
 }
 
+func (o *Options) IfPodIsFailedShareLogs(pod *corev1.Pod, previewNamespace string) error {
+	if pod.Status.Phase == corev1.PodFailed {
+		log.Logger().Infof("found pod %s in namespace %s in state %s", pod.Name, previewNamespace, pod.Status.Phase)
+
+		logs := o.KubeClient.CoreV1().Pods(previewNamespace).GetLogs(pod.Name, &corev1.PodLogOptions{})
+		stream, err := logs.Stream(context.Background())
+		if err != nil {
+			return err
+		}
+		defer stream.Close()
+
+		scanner := bufio.NewScanner(stream)
+
+		var m string
+		for scanner.Scan() {
+			m += scanner.Text()
+		}
+		return fmt.Errorf("failed pod %s in namespace %s: %s", pod.Name, previewNamespace, m)
```

Comment:
This looks functionally correct to me. But I am worried about the memory consumption. The logs can be large... It would be better to do it like in the other place you pointed out and log after each iteration. But for the output to make sense then the code would need to be restructured a bit so the helmfile error is logged before.

Reply:
It's not going to be a lot of logs, I've just pulled the logs of a service we have, that's very busy (with a heartbeat that's also logged), it's been running for 3 days, the logs only equate to 5MB. 

In this scenario, the pod has only just started and it has failed to start, therefore we can assume that the size of the logs is going to be much, much smaller than this. So I think this is fine, this is a feature that will only trigger if there is a helmfile error, which the previous response was useless. I think we're fine to proceed as is.

We'll use it in our environment with all of our services, so we'll hit this problem pretty quickly if there is one.

Pre-optimisation is the root of all evil ;)

Reply:
So while it is unnecessarily wasteful it will probably work, yes.

If you call that optimisation you haven't seen what optimisation is...